conda_smithy/cli.py

```diff
@@ -257,6 +260,22 @@ def __init__(self, parser):
             action="append",
             help="drone server URL to register this repo. multiple values allowed",
         )
+        scp.add_argument(
+            "--cirun-resources",
```

Comment:
Is this required when `cirun` is enabled? If optional, we need to set the default value to an iterable (not `None`), because otherwise we hit this bug:

```
conda-smithy register-ci --feedstock_directory . --organization conda-forge --without-azure --without-travis --without-circle --without-appveyor --without-drone --without-webservice --without-anaconda-token 
No circle token.  Create a token at https://circleci.com/account/api and
put it in ~/.conda-smithy/circle.token
No appveyor token. Create a token at https://ci.appveyor.com/api-token and
Put one in ~/.conda-smithy/appveyor.token
No drone token. Create a token at https://cloud.drone.io/account and
Put one in ~/.conda-smithy/drone.token
No anaconda token. Create a token via
  anaconda auth --create --name conda-smithy --scopes "repos conda api"
and put it in ~/.conda-smithy/anaconda.token
WARNING: No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.22
Adding in variants from internal_defaults
Adding in variants from /Users/jrodriguez/devel/cf-autotick-bot-test-package-feedstock/recipe/conda_build_config.yaml
CI Summary for conda-forge/cupy-feedstock (can take ~30s):
Warning: By not registering an Anaconda/Binstar tokenyour feedstock CI may not be able to upload packagesto anaconda.org by default. It is recommended to set`upload_packages: False` per provider field inconda-forge.yml to disable package uploads.
Travis registration disabled.
Circle registration disabled.
Azure registration disabled.
Appveyor registration disabled.
Drone registration disabled.
Traceback (most recent call last):
  File "/Users/jrodriguez/.local/anaconda/envs/smithy-dev/bin/conda-smithy", line 33, in <module>
    sys.exit(load_entry_point('conda-smithy', 'console_scripts', 'conda-smithy')())
  File "/Users/jrodriguez/devel/conda-smithy/conda_smithy/cli.py", line 712, in main
    args.subcommand_func(args)
  File "/Users/jrodriguez/devel/conda-smithy/conda_smithy/cli.py", line 386, in __call__
    for resource in args.cirun_resources:
TypeError: 'NoneType' object is not iterable
```

If required, then we need to error out properly.